src/frontend/components/StudentLiveControlButtons/StudentJoinDiscussionButton/index.tsx

```diff
@@ -0,0 +1,187 @@
+    description:
+      'Title displayed in the layer to ask the user to choose a username in order to connect to the chat',
+    id: 'components.StudentJoinDiscussionButton.chooseUsername',
+  },
+  UsernameAlreadyExists: {
+    defaultMessage: 'Username already exists, please choose another one',
+    description:
+      'Error message shown when the connection to the chat failed if the username is already in use.',
+    id: 'components.StudentJoinDiscussionButton.usernameAlreadyExists',
+  },
+  username: {
+    defaultMessage: 'Username',
+    description: 'label used by the input text to write the username to choose',
+    id: 'components.StudentJoinDiscussionButton.username',
+  },
+  confirm: {
+    defaultMessage: 'Confirm',
+    description: 'Label used by the confirm button',
+    id: 'components.StudentJoinDiscussionButton.confirm',
+  },
+  cancel: {
+    defaultMessage: 'Cancel',
+    description: 'Label used by the cancel button',
+    id: 'components.StudentJoinDiscussionButton.cancel',
+  },
+});
+
+export const StudentJoinDiscussionButton = () => {
+  const intl = useIntl();
+  const [username, setUsername] = useState('');
+  const [showUsernameForm, setShowUsernameForm] = useState(false);
+  const {
+    asked,
+    accepted,
+    rejected,
+    usernameAlreadyExisting,
+    reset,
+    setAsked,
+  } = useParticipantWorkflow((state) => ({
+    asked: state.asked,
+    accepted: state.accepted,
+    rejected: state.rejected,
+    usernameAlreadyExisting: state.usernameAlreadyExisting,
+    setAsked: state.setAsked,
+    reset: state.reset,
+  }));
+
+  const closeUserNameForm = () => {
+    setUsername('');
+    setShowUsernameForm(false);
+    reset();
+  };
+
+  const askToJoinDiscussion = async () => {
+    try {
+      //  alert teacher for the request through XMPP
+      await converse.askParticipantToJoin(username);
+    } catch (error) {
+      //  when failling to alert the teacher
+      setShowUsernameForm(true);
+      return;
+    }
+
+    //  save the request in the store
+    setAsked();
+    //  hide the form
+    setShowUsernameForm(false);
+  };
+
+  useEffect(() => {
+    if (usernameAlreadyExisting) {
+      setShowUsernameForm(true);
+    }
+  }, [usernameAlreadyExisting]);
+
+  //  if user is rejected : alert user and reset state
+  useEffect(() => {
+    if (rejected) {
+      reset();
+      toast.error(intl.formatMessage(messages.rejected));
+    }
+  }, [rejected, reset]);
+
+  if (accepted) {
+    return <Redirect to={PUBLIC_JITSI_ROUTE()} />;
+  }
+
+  if (asked) {
+    return (
+      <Box align="center">
+        <WaitingJoinDiscussionSVG
+          containerStyle={{ width: '54px', height: '54px', background: 'none' }}
+          iconColor={'blue-active'}
+        />
+        <Text
+          color={normalizeColor('blue-active', theme)}
+          margin={{ top: '6px' }}
+          size="12px"
+        >
+          {intl.formatMessage(messages.waitingInstructor)}
```

Comment:
In a JSX context, component [`<FormattedMessage />`](https://formatjs.io/docs/react-intl/components/#formattedmessage) is more appropriate than `{intl.formatMessage(...` from my point of view.